app/components/bali/heatmap/component.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Bali
+  module Heatmap
+    class Component < ApplicationViewComponent
+      attr_reader :title, :data, :width, :height, :subtitle, :gradient_colors
+
+      renders_one :hovercard_title, ->(text) { tag.p(text, class: 'title is-7 mb-0') }
+      renders_one :legend_title, ->(text) { tag.p(text, class: 'title is-7') }
+      renders_one :y_axis_title, ->(text) { tag.p(text, class: 'title is-7 pb-3') }
+      renders_one :x_axis_title, ->(text) {
+                                   tag.p(text, class: 'title is-7 mt-2 has-text-centered')
+                                 }
+
+      def initialize(width: 480, height: 480, data: {}, **options)
+        @width = width
+        @height = height
+        @data = data
+        @title = options.delete(:title)
+        @subtitle = options.delete(:subtitle)
+        @gradient_base_color = options.delete(:gradient_base_color) || '#008806'
+        @gradient_colors = Bali::Utils::ColorPicker.new.gradient(@gradient_base_color)
+      end
```

Comment:
Un par de cosas:
1. Simplificar el nombre del gradient_base_color a color, ya que no se puede confundir con otro tipo de color.
2. Pasar las options para poderle pasar un class o id al root tag.

```suggestion
      def initialize(width: 480, height: 480, color: '#008806', data: {}, **options)
        @width = width
        @height = height
        @data = data
        @color = color
        @gradient_colors = Bali::Utils::ColorPicker.new.gradient(@color)
        
        @options = prepend_class_name(options, 'heatmap-component table-container')
      end
```